posts/2022-04-12-java18-22004.adoc

```diff
@@ -0,0 +1,218 @@
+* link:https://openjdk.java.net/jeps/417[Vector API (Third Incubator)]
+* link:https://openjdk.java.net/jeps/418[Internet-Address Resolution SPI]
+* link:https://openjdk.java.net/jeps/419[Foreign Function & Memory API (Second Incubator)]
+* link:https://openjdk.java.net/jeps/420[Pattern Matching for switch (Second Preview)]
+* link:https://openjdk.java.net/jeps/421[Deprecate Finalization for Removal]
+
+[NOTE] 
+====
+Java Security support has been removed with Java 18+ in Liberty.  If "websphere.java.security" is set in `bootstrapping.properties`, it will produce the following error in the log:
+```
+CWWKE0955E: Java Security was requested in bootstrapping.properties while using JDK "18", however this option is no longer valid when using Java 18 and later.
+```
+====
+
+To use Java 18 with Liberty, link:https://jdk.java.net/18/[download] the general availibility release, <<run, install 22.0.0.4>> version of Open Liberty, and edit your Liberty server's link:{url-prefix}/docs/latest/reference/config/server-configuration-overview.html#server-env[server.env file] to point `JAVA_HOME` to your Java 18 installation.
+
+For more information on Java 18, please visit the Java 18 link:https://jdk.java.net/18/release-notes[release notes page], link:https://download.java.net/java/early_access/jdk18/docs/api/[API Javadoc page] or link:https://jdk.java.net/18/[download page].
+For more information on Open Liberty, please visit our link:{url-prefix}/docs[documentation page].
+
+[#certificate]
+== Support client certificates with URL encoded strings
+
+An `X.509` certificate is a digital certificate that is used to manage identity and security in internet communications and computer networking.
+
+Intermediate servers, such as a web server, proxy server, or load balancer can send X.509 certificates to Open Liberty servers. Due to the deprecation of long lines folding in the `HTTP` header continuation, some open source intermediate servers might send a client certificate in the Privacy-Enhanced Mail (`PEM`) url-encoded format.
+
+Open Liberty 22.0.0.4 adds the ability to auto detect and process this new format.
+
+An example of a truncated url-encoded certificate sent to application server:app-name:
+
+[source, role="no_copy"]
+----
+[-----BEGIN%20CERTIFICATE-----%0AMIIDODCCAiCgAwIBAgIIHbO76YfCH1cwDQYJKoZIhvcNAQELBQAw
+OjELMAkGA1UE%0ABhMCVVMxDDAKBgNVBAoTA0lCTTEMMAoGA1UECxMDVFdTMQ8wDQYDVQQDEwZDbGll%0AbnQ
+wHhcNMTUxMTAzMTYyODM1WhcNMzUxMDMwMTYyODM1WjA6MQswCQYDVQQGEwJV%0AUzEMMAoGA1UEChMDSUJNM
+QwwCgYDVQQLEwNUV1MxDz%2F%2Bpo%2FF%2Bzi%2F0sfAUwRdfgMm%2FAKAGzwMQufOYeKCgMULtq14QAJQnL
+mq4M%2FM00%0AC5QyeYtdaTou%2BMsLmoa1tkq2VSDVxAcktJyRSRsox36G7EHDLV4U2gtR6xczNEjw%0A2%2
+Bsj772FjdAMXRSR%0A-----END%20CERTIFICATE-----%0A]
+----
+
+Note: 
+
+* (1) The header (i.e BEGIN ) and footer (i.e END) information are appended before sending to application server.
+* (2) The encoded character (i.e `%`) presents in the certificate.
```

Comment:
what does this mean? presents what in the certificate? Does this refer specifically to `%` ? The statement suggests that the percent symbol is just one example of an encoded character but there's not enough information to understand the context or why this is important to note

Reply:
@pmd1nh would you be able to clarify this for us so that we can make the appropriate change? Thanks in advance :)

Reply:
If the certificate string contains any % character, it indicates that the string is encoded.